### PR TITLE
lint: Make command line arguments accessible to helper functions

### LINKED ISSUE #12740: Format checking/fixing script breaks with python3.8
Description:

Running `tools/code_format/check_format.py check` (or with `fix`) with python3.8.0 or python3.8.7 on MacOS fails with the following error:

```
➜  envoy git:(master) tools/code_format/check_format.py check
multiprocessing.pool.RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/Users/lisalu/.pyenv/versions/3.8.0/lib/python3.8/multiprocessing/pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
  File "/Users/lisalu/src/envoy/tools/code_format/check_format.py", line 965, in checkApiShadowStarlarkFiles
    if operation_type == "check":
NameError: name 'operation_type' is not defined
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "tools/code_format/check_format.py", line 1152, in <module>
    error_messages += sum((r.get() for r in results), [])
  File "tools/code_format/check_format.py", line 1152, in <genexpr>
    error_messages += sum((r.get() for r in results), [])
  File "/Users/lisalu/.pyenv/versions/3.8.0/lib/python3.8/multiprocessing/pool.py", line 768, in get
    raise self._value
NameError: name 'operation_type' is not defined
```

While the argument is correctly parsed into the `operation_type` variable in the main function it is undefined/not visible to the workers running checkFormat. I am guessing this behavior changed with Python3.8 (https://bugs.python.org/issue39931?) as I am able to run the script successfully with Python3.7.8 and older versions.

### PR DESCRIPTION
**Pull request by LisaLudique**
Commit Message: Make command line arguments accessible to helper functions in formatting script
Additional Description: The formatting script does not recognize the variables from the command line arguments when running Python >=3.8 on MacOS due to changes in multiprocessing behavior in the new Python version (https://bugs.python.org/issue39931). This change passes in the command line arguments so they are accessible by the helper functions in which they are needed.
Risk Level: Low
Testing: Existing unit tests
Docs Changes: N/A
Release Notes: N/A
Fixes #12740

### CODE REVIEW

File: tools/code_format/check_format.py [hunk 1]
```diff
@@ -916,25 +925,25 @@ def checkFormat(file_path):
...
+      error_messages += fixSourcePath(file_path, command_line_args)
+    error_messages += checkSourcePath(file_path, command_line_args)
 
   if error_messages:
     return ["From %s" % file_path] + error_messages
   return error_messages
 
 
-def checkFormatReturnTraceOnError(file_path):
+def checkFormatReturnTraceOnError(file_path, command_line_args):
```
Comment (htuch) [inline tools/code_format/check_format.py:943]: What do you think about making this a class and have command line args be a member variable? That could reduce the amount of threading required.

Comment (LisaLudique) [reply tools/code_format/check_format.py:943]: This sounds like a better approach. I'll get to this this week.

Comment (LisaLudique) [reply tools/code_format/check_format.py]: Sorry for the delay, this is now ready for another look

File: tools/code_format/check_format.py [hunk 2]
```diff
@@ -1110,7 +1059,7 @@ def ownedDirectories(error_messages):
     owned = []
     maintainers = [
         '@mattklein123', '@htuch', '@alyssawilk', '@zuercher', '@lizan', '@snowp', '@asraa',
-        '@yavlasov', '@junr03', '@dio', '@jmarantz'
+        '@yavlasov', '@junr03', '@dio', '@jmarantz', '@antoniovicente'
```
Comment (htuch) [inline tools/code_format/check_format.py:1062]: Thanks for this update. For review benefit, are there any other sites that aren't refactors or CLI plumbing related?

**Review by htuch — COMMENTED** [review]
This is really nice, thanks so much for the refactor, this will be way more maintainable! QQ on what changes have happened, it's kind of hard to understand due to GitHub's diff view being linear.

File: tools/code_format/check_format.py (hunk 2 above)
Comment (LisaLudique) [reply tools/code_format/check_format.py:1062]: No worries, the diff is hard to read.

I defined a FormatChecker class and moved all the functions in https://github.com/envoyproxy/envoy/blob/130c7c4e271fe306ae4dd747daa5f09ff31aef79/tools/code_format/check_format.py#L186-L1030 under it. I moved these arguments https://github.com/envoyproxy/envoy/blob/130c7c4e271fe306ae4dd747daa5f09ff31aef79/tools/code_format/check_format.py#L1083-L1098 to the initializer.

That particular highlighted change to the owners list is possibly due to my needing to rebase?

**Review by htuch — APPROVED** [review]
LGTM, thanks!

**Pull request merged**